core/src/main/clojure/xtdb/indexer/live_index.clj

```diff
@@ -319,14 +319,21 @@
                                                           (catch Exception _
                                                             (throw (.exception ^StructuredTaskScope$Subtask %)))))))
                                    (util/rethrowing-cause))]
-
-            (let [table-block-paths (table-cat/finish-block! table-cat block-idx table-metadata)]
-              (.finishBlock block-cat block-idx latest-completed-tx table-block-paths))
-
             (let [added-tries (for [[table-name {:keys [trie-key data-file-size]}] table-metadata]
-                                (trie-cat/->added-trie table-name trie-key data-file-size))]
+                                (trie/->added-trie table-name trie-key data-file-size))]
               (.addTries trie-cat added-tries)
-              @(.appendMessage log (Log$Message$TriesAdded. added-tries))))))
+              @(.appendMessage log (Log$Message$TriesAdded. added-tries)))
+
+            (let [all-tables (set (concat (keys table-metadata) (.getAllTableNames block-cat)))
+                  table->current-tries (->> all-tables
+                                            (map (fn [table-name]
+                                                   (MapEntry/create table-name
+                                                                    (->> (trie-cat/trie-state trie-cat table-name)
+                                                                         trie-cat/current-tries))))
```

Comment:
I suspect we want all the tries stored in here, live and garbage - idea being to reflect what's in the object store. Maybe not an issue now, but when we add GC to the LSM we'll want to know all the tries, not just the live ones.